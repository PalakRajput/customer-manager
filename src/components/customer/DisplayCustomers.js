import React, { useEffect } from 'react'
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux/';
import { deleteCustomer, getCustomerAction } from '../../redux/actions/CustomerActions';



const DisplayCustomers = () => {
    const dispatch = useDispatch();
    const customers = useSelector((state) => state.customer.customers);
    //const successMessage = useSelector(state => state.customer.successMessage)
    useEffect(() => {
        dispatch(getCustomerAction())
    }, [dispatch])
    const handleDeleteClick = (id) => {
        dispatch(deleteCustomer(id))
    }
    return (
        <>
            <div style={{ margin: '20px' }}>


                {customers.length !== 0 ? (customers.map(customer => (
                    <React.Fragment key={customer.id}>
                        <div className='row'>
                            <div className='col md-4 offset-3' >
                                <div className='card cardShadow' >
                                    <div className='card-body cardColor' >
                                        <div className='card-title' ><Link to='/' state={{ customer }}>{customer.fname + " " + customer.lname}</Link></div>
                                        {customer.city}
                                        <br />
                                        <Link to={`/orders/${customer.id}`} >View {customer.ordersList.length} Orders</Link>
                                    </div>
                                    <div className='card-footer'>
                                        <button className='btn btn-danger btn-small' onClick={() => handleDeleteClick(customer.id)}>Delete</button>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <br />
                    </React.Fragment>
                ))
                ) : (
                    <div>No Customers to Display. Please add customers</div>
                )
                }



            </div>
            {/* <ToastContainer autoClose={2000}/> */}
        </>
    )
}
export default DisplayCustomers;
