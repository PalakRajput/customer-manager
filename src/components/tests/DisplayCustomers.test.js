import { render, screen } from '@testing-library/react'
import React from 'react'
import DisplayCustomers from '../customer/DisplayCustomers'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
const store = mockStore({
    customer:{
        customers:[],
        successMessage: "",
        error:""
    }
})

describe("Display Customers", () => {
    it (" should render correctly ", () => {
           const {asFragment} =  render(<Provider store={store}><DisplayCustomers /></Provider>)
           expect(asFragment(<DisplayCustomers />)).toMatchSnapshot();
    })
})