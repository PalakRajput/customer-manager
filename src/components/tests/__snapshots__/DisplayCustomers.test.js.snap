// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Customers  should render correctly  1`] = `
<DocumentFragment>
  <div
    style="margin: 20px;"
  >
    <div>
      No Customers to Display. Please add customers
    </div>
  </div>
</DocumentFragment>
`;
